index.html

```diff
@@ -300,85 +300,14 @@ <h3>
         </h3>
         <p>
           The {{Document/onvisibilitychange}} attribute is an <a>event handler
-          IDL attribute</a> for the `"visibilitychange"` event type (see
-          [[[#reacting-to-visibilitychange]]]).
+          IDL attribute</a> for the "[=Document/visibilitychange=]" event type.
+        </p>
+
+        <p class="note">
+          The "[=Document/visibilitychange=]" event is invoked from the [[HTML]] spec.
         </p>
       </section>
     </section>
-    <section>
-      <h2 id="reacting-to-visibilitychange">
-        Reacting to `"visibilitychange"` changes
-      </h2>
-      <p>
-        The <a>task source</a> for these <a>tasks</a> is the <a>user
-        interaction task source</a>.
-      </p>
-      <p>
-        When the user agent determines that the visibility of the {{Document}}
-        of the <a>top-level browsing context</a> has changed, the user agent
-        MUST run the following steps:
-      </p>
-      <ol class="algorithm">
-        <li>Let |doc:Document| be the {{Document}} of the <a>top-level browsing
-        context</a>.
-        </li>
-        <li>If |doc| is now [=Document/visible=]:
-          <ol>
-            <li>If traversing to a <a>session history entry</a>, run the <a>now
-            visible algorithm</a> before running the step to fire the
-            [=Window/pageshow=] event.
-            </li>
-            <li>Otherwise, <a>queue a task</a> that runs the <a>now visible
-            algorithm</a>.
-            </li>
-          </ol>
-        </li>
-        <li>Else if |doc| is now not [=Document/visible=], or if the user agent
-        is to [=Window/unload=] |doc|:
-          <ol>
-            <li>If the user agent is to [=Window/unload=] the {{Document}}, run
-            the <a>now hidden algorithm</a> during the <a>unloading document
-            visibility change steps</a>.
-            </li>
-            <li>Otherwise, <a>queue a task</a> that runs the <a>now hidden
-            algorithm</a>.
-            </li>
-          </ol>
-        </li>
-      </ol>
-      <p>
-        The <dfn data-export="" data-dfn-type="abstract-op">now visible
-        algorithm</dfn> runs the following steps synchronously:
-      </p>
-      <ol class="algorithm">
-        <li>Let |doc:Document| be the {{Document}} of the <a>top-level browsing
-        context</a>.
-        </li>
-        <li>
-          <a>Fire an event</a> named "`visibilitychange`" that bubbles, isn't
-          cancelable, and has no default action, at the |doc|.
-        </li>
-        <li>Run <dfn data-export="" data-dfn-type="abstract-op">external now
```

Comment:
I vaguely remember other specifications relying on this. Worthwhile to dig through past commits to see why we added this.

Reply:
The comments on this issue say otherwise, but it's possible...
I think the discussion here went in the direction of using direct calls rather than hooks, I tend to support that - much more straightforward.